File: backend/Ecommerce.Domain/src/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Ecommerce.Domain.src.Entities
{
    public class Product : BaseEntity
    {
        public string Title { get; set; } = default!;
        public string Description { get; set; } = default!;
        public float Price { get; set; }
        public Category Category { get; set; }
        public int Inventory { get; set; }
        public ProductImage Image { get; set; } = default!;
    }

    public enum Category
    {
        Vegetables,
        Meat,
        Dairy,
        Others
    }
}

File: frontend/src/types/Product.ts
import { Guid } from "guid-typescript";
import { Category } from "./Category";
import { Image } from "./Image";

export interface Product {
    id: Guid;
    title: string;
    description: string;
    price: number;
    category: number;
    inventory: number;
    image: Image
}

export interface ProductCreateDto {
    title: string;
    description: string;
    price: number;
    category: number;
    inventory: number;
    image: Image
}

export interface ProductUpdateDto {
    id: Guid;
    update: {
        title: string;
        description: string;
        price: number;
        category: number;
        inventory: number;
        image: Image
    }
}